lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -175,8 +175,20 @@ public function fetchAll($fetchMode = null, $columnIndex = 0)
             return $rows;
         }
 
-        foreach ($rows as $num => $row) {
-            $rows[$num] = $this->fixRow($row, $iterateRow, $fixCase);
+        if ($fetchMode === PDO::FETCH_COLUMN) {
```

Comment:
@morozov can you isolate the contents of the `if` and `else` statements into private methods? Early return preferred